checkov/terraform/plan_parser.py

```diff
@@ -121,31 +121,69 @@ def _prepare_resource_block(
 ) -> List[Dict[str, Dict[str, Any]]]:
+    """ Find all child modules if any. Including any amount of nested child modules.
+
+    :param child_modules: list of terraform child_module objects
+    :param resource_changes: a resource address to resource changes dict
+    :param root_module_conf: configuration block of the root module
+    :returns:
+        list of terraform resource blocks
     """
-    Find all child modules if any. Including any amount of nested child modules.
-    :type: child_modules: list of tf child_module objects
-    :rtype: resource_blocks: list of hcl resources
-    """
+
     resource_blocks = []
     for child_module in child_modules:
-        if child_module.get("child_modules", []):
-            nested_child_modules = child_module.get("child_modules", [])
-            nested_blocks = _find_child_modules(nested_child_modules, resource_changes)
+        nested_child_modules = child_module.get("child_modules", [])
+        if nested_child_modules:
+            nested_blocks = _find_child_modules(
+                child_modules=nested_child_modules,
+                resource_changes=resource_changes,
+                root_module_conf=root_module_conf
+            )
             for resource in nested_blocks:
                 resource_blocks.append(resource)
+
+        module_address = child_module.get("address", "")
+        module_call_resources = _get_module_call_resources(
+            module_address=module_address,
+            root_module_conf=root_module_conf,
+        )
+
         for resource in child_module.get("resources", []):
+            module_call_conf = None
+            if module_address and module_call_resources:
+                module_call_conf = next(
+                    (
+                        module_call_resource
+                        for module_call_resource in module_call_resources
+                        if f"{module_address}.{module_call_resource['address']}" == resource["address"]
+                    ),
+                    None
+                )
+
             resource_block, prepared = _prepare_resource_block(
                 resource=resource,
-                conf=None,
+                conf=module_call_conf,
                 resource_changes=resource_changes,
             )
             if prepared is True:
                 resource_blocks.append(resource_block)
     return resource_blocks
 
 
+def _get_module_call_resources(module_address: str, root_module_conf: dict[str, Any]) -> list[dict[str, Any]]:
+    """Extracts the resources from the 'module_calls' block under 'configuration'"""
+
+    for module_name in module_address.split("."):
+        if module_name == "module":
+            # module names are always prefixed with 'module.', therefore skip it
+            continue
+        root_module_conf = root_module_conf.get("module_calls", {}).get(module_name, {}).get("module", {})
```

Comment:
@gruebel  Do we expect that `module_address.split(".")` will yield only one module name? (besides "module" itself)?

Reply:
technically the `split()` method returns an array of at least one entry. If it is not a module, then the whole function will return an empty list. if it is indeed a module or even a level x nested module, then the for loop will dig deeper and deeper through the module calls specific blocks under the root configuration block and return a list of resources. I hope this explains it a bit better 🙂

Reply:
Can a module address be in a format other than "module.module_name" ? 
If it can also be "module.module_name.other_stuff" for example, then going deeper won't necessarily get the correct results.

I think the function implicitly expects module address of the format "module.module_name" and maybe it will be worth enforcing it

Reply:
due to Terraforms way of creating addresses for modules, it is expected to be `module.module_name` and for nested modules `module.module_name.module.xxx.module.xxx` etc.

have to think about, if there is an elegant way to "enforce" it 🤔

Reply:
If the addresses are created from a 3rd party library then probably that's not as critical, right?

I thought that we had control over that

Reply:
sadly no, we are just trying to extract information, which Terraform gives us 😄